src/Sentry.Unity/Il2CppEventProcessor.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Sentry.Extensibility;
+using Sentry.Protocol;
+
+namespace Sentry.Unity
+{
+    // TODO: Make sure this whole functionality/class is only compiled when:
+    // * Compiling for the il2cpp backend.
+    // * Using Unity 2020 or later, as we use internal `libil2cpp` APIs that are
+    //   only available there.
```

Comment:
One way to do this is have the code be compiled by users instead of shipping in a .dll - that's what I've suggested here: https://github.com/getsentry/sentry-unity/pull/768/files

Reply:
We have an alternative proposal using delegates and the dependecy injection. Maybe we discuss that on a call